specification/security/resource-manager/Microsoft.Security/stable/2024-08-01/customRecommedations.json

```diff
@@ -0,0 +1,473 @@
+          ],
+          "x-ms-enum": {
+            "name": "securityIssue",
+            "modelAsString": true
+          },
+          "x-ms-client-default": "BestPractices"
+        },
+        "metadata": {
+          "$ref": "#/definitions/CustomRecommendationMetaData"
+        }
+      }
+    },
+    "CustomRecommendationRequestBody": {
+      "type": "object",
+      "description": "Custom Recommendation request body",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "describes Custom Recommendation request body properties.",
+          "$ref": "#/definitions/CustomRecommendationRequestBodyProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "CustomRecommendationRequestBodyProperties": {
+      "type": "object",
+      "description": "describes the Custom Recommendation request body properties",
+      "properties": {
+        "query": {
+          "description": "KQL query representing the Recommendation results required.",
+          "type": "string"
+        },
+        "cloudProviders": {
+          "description": "List of all standard supported clouds.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RecommendationSupportedCloud"
+          }
+        },
+        "severity": {
+          "description": "The severity to relate to the assessments generated by this Recommendation.",
+          "type": "string",
+          "enum": [
+            "High",
+            "Medium",
+            "Low"
+          ],
+          "x-ms-enum": {
+            "name": "severityEnum",
+            "modelAsString": true
+          },
+          "x-ms-client-default": "Low"
+        },
+        "securityIssue": {
+          "description": "The severity to relate to the assessments generated by this Recommendation.",
+          "type": "string",
+          "enum": [
+            "Vulnerability",
+            "ExcessivePermissions",
+            "AnonymousAccess",
+            "NetworkExposure",
+            "TrafficEncryption",
+            "BestPractices"
+          ],
+          "x-ms-enum": {
+            "name": "securityIssue",
+            "modelAsString": true
+          },
+          "x-ms-client-default": "BestPractices"
+        },
+        "displayName": {
+          "description": "The display name of the assessments generated by this Recommendation.",
+          "type": "string"
+        },
+        "description": {
+          "description": "The description to relate to the assessments generated by this Recommendation.",
+          "type": "string"
+        },
+        "remediationDescription": {
+          "description": "The remediation description to relate to the assessments generated by this Recommendation.",
+          "type": "string"
+        }
+      }
+    },
+    "CustomRecommendationMetaData": {
```

Comment:
>CustomRecommendationMetaData [](http://example.com/codeflow?start=5&length=28)

These appear to duplicate the top-level systemData. Can you eliminate these properties and just use the systemData?

Reply:
Done